wagtail/snippets/views/snippets.py

```diff
@@ -174,17 +171,25 @@ def get_base_queryset(self):
             self.queryset = self.queryset(self.request)
         return super().get_base_queryset()
 
-    def _get_title_column(self, field_name, column_class=SnippetTitleColumn, **kwargs):
-        # Use SnippetTitleColumn class to use custom template
-        # so that buttons from snippet_listing_buttons hook can be rendered
-        return super()._get_title_column(field_name, column_class, **kwargs)
-
     def get_columns(self):
         return [
             BulkActionsCheckboxColumn("checkbox", accessor=lambda obj: obj),
             *super().get_columns(),
         ]
 
+    def get_list_dropdown_buttons(self, instance, parent_context):
+        buttons = super().get_list_dropdown_buttons(instance, parent_context)
+        next_url = self.request.path
+        button_hooks = hooks.get_hooks("register_snippet_listing_buttons")
+
+        for hook in button_hooks:
+            buttons.extend(hook(instance, self.request.user, next_url))
+
+        for hook in hooks.get_hooks("construct_snippet_listing_buttons"):
+            hook(buttons, instance, self.request.user, parent_context)
```

Comment:
As mentioned in the PR description, the `context` passed to the [`construct_snippets_listing_buttons`](https://docs.wagtail.org/en/stable/reference/hooks.html#construct-snippet-listing-buttons) hook is now the context that has been reduced to the tables framework scope (`parent_context` in the `Column` component will be the context of the `Table` component), which means most context variables from the view are not available. If this is not desirable, I can drop the last commit that does this refactoring.

Reply:
I'm inclined to remove it too, but this does mean people will need to change the signature of their hooks to not accept the fourth argument. Do you think it's something we should handle gracefully with `try`/`except` and raising a warning, or do you think an upgrade consideration telling people to remove the fourth argument is enough?

Reply:
Done in eaadd81. Tests and docs have been added in later commits.